### PR TITLE
Add javadoc for setting resources in certificate builders.

### PR DESCRIPTION
**Pull request by lolepezy**

### CODE REVIEW

**Comment by ties** [comment]
I like these doc fixes.

**Review by ties — APPROVED** [review]

File: src/main/java/net/ripe/rpki/commons/crypto/x509cert/X509CertificateBuilderHelper.java [hunk 1]
```diff
@@ -78,6 +78,19 @@
  * Because we want to maintain the pattern where a specific Certificate builder
  * can be chained like: builder.withValidity(val).withSubjectDn(subject) etc...
  * dynamic typing would be required.. hence delegation.
+ *
+ * Even though RPKI is all about resource certificates, there is not resource-specific
+ * version if this builder helper as it is not so easy to make sure that resources are
```
Comment (bjpbakker) [inline src/main/java/net/ripe/rpki/commons/crypto/x509cert/X509CertificateBuilderHelper.java:83]:
```suggestion
 * version of this builder helper as it is not so easy to make sure that resources are
```

File: src/main/java/net/ripe/rpki/commons/crypto/x509cert/X509CertificateBuilderHelper.java [hunk 2]
```diff
@@ -78,6 +78,19 @@
...
+ * set in a compile-time guaranteed fashion. Resources can be set
+ *
+ *  - using withResources;
+ *  - using withInheritedResourceTypes.
+ *
+ * There is complicated relationships between these two ways of setting resources
```
Comment (bjpbakker) [inline src/main/java/net/ripe/rpki/commons/crypto/x509cert/X509CertificateBuilderHelper.java:89]:
```suggestion
 * There are complicated relationships between these two ways of setting resources
```

File: src/main/java/net/ripe/rpki/commons/crypto/x509cert/X509CertificateBuilderHelper.java [hunk 3]
```diff
@@ -78,6 +78,19 @@
...
+ * (@see X509CertificateBuilderHelper.validateResource)
+ * which results in a inevitable runtime check. Thus moving resource setting
```
Comment (bjpbakker) [inline src/main/java/net/ripe/rpki/commons/crypto/x509cert/X509CertificateBuilderHelper.java:91]:
```suggestion
 * which results in an inevitable runtime check. Thus moving resource setting
```

File: src/main/java/net/ripe/rpki/commons/crypto/x509cert/X509CertificateBuilderHelper.java [hunk 4]
```diff
@@ -78,6 +78,19 @@
...
+ * into a separate resource-specialised builder doesn't make sense.
```
Comment (bjpbakker) [inline src/main/java/net/ripe/rpki/commons/crypto/x509cert/X509CertificateBuilderHelper.java:92]:
```suggestion
 * into a separate resource-specialized builder doesn't make sense.
```

**Review by bjpbakker — APPROVED** [review]

File: src/main/java/net/ripe/rpki/commons/crypto/x509cert/X509CertificateBuilderHelper.java (hunk 4 above)
Comment (lolepezy) [reply src/main/java/net/ripe/rpki/commons/crypto/x509cert/X509CertificateBuilderHelper.java:92]: We use en_uk I guess.

**Pull request merged**